src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -580,15 +579,10 @@ private static int GetIntQueryParameterValue(NameValueCollection queryStringName
                 IDurableOrchestrationClient client = this.GetClient(request);
 
                 object input = null;
-                if (request.Content != null)
+                if (request.Content?.Headers?.ContentLength > 0)
                 {
-                    using (Stream s = await request.Content.ReadAsStreamAsync())
-                    using (StreamReader sr = new StreamReader(s))
-                    using (JsonReader reader = new JsonTextReader(sr))
-                    {
-                        JsonSerializer serializer = JsonSerializer.Create(MessagePayloadDataConverter.MessageSettings);
-                        input = serializer.Deserialize<object>(reader);
-                    }
+                    string json = await request.Content.ReadAsStringAsync();
+                    input = JsonConvert.DeserializeObject(json, MessagePayloadDataConverter.MessageSettings);

```

Comment:
I don't think we should pivot to `JObject.LoadAsync()`. As you mentioned, we have no way to utilize our `JsonSerializationSettings` then. I'm also not sure how well the type coercion works under the hood from `JObject` to the customer's specified types in their code (I'm assuming well, but I prefer the notion of leaving it an `object`).

What we really want is `serializer.DeserializeAsync()`, but that only exists in net core 3.0, not even in .net standard 2.1. Regardless, at the version we are stuck with, I highly doubt we will get access to it. 

In general, the main downside here is the [memory usage](https://www.newtonsoft.com/json/help/html/Performance.htm). Since this is only used in orchestration starts, and in general customers will hit large IO costs when using objects > 64 KB anyways (due to having to store the message in blobs instead of queues), I am not too concerned about this currently. We can always revisit this later if it becomes an issue.